src/helix-ui/elements/HXModalElement.js

```diff
@@ -37,50 +37,31 @@ export class HXModalElement extends HXElement {
         return `<style>${shadowStyles}</style>${shadowMarkup}`;
     }
 
-    static get observedAttributes () {
+    static get $observedAttributes () {
         return [ 'open' ];
     }
 
-    constructor () {
-        super();
-        this._close = this._close.bind(this);
-        this._keyUp = this._keyUp.bind(this);
+    $onCreate () {
+        this._onBtnClose = this._onBtnClose.bind(this);
+        this._onKeyUp = this._onKeyUp.bind(this);
     }
 
-    connectedCallback () {
+    $onConnect () {
         this.$upgradeProperty('open');
         this.setAttribute('aria-hidden', !this.open);
-
-        this._btnClose.addEventListener('click', this._close);
-        document.addEventListener('keyup', this._keyUp);
+        this._btnClose.addEventListener('click', this._onBtnClose);
     }
 
-    disconnectedCallback () {
-        this._btnClose.removeEventListener('click', this._close);
-        document.removeEventListener('keyup', this._keyUp);
+    $onDisconnect () {
+        this._btnClose.removeEventListener('click', this._onBtnClose);
     }
 
-    attributeChangedCallback (attr, oldVal, newVal) {
-        let isOpen = (newVal !== null);
-        this.setAttribute('aria-hidden', !isOpen);
-
-        if (newVal !== oldVal) {
-            this.$emit(isOpen ? 'open' : 'close');
+    $onAttributeChange (attr, oldVal, newVal) {
+        if (attr === 'open') {
+            this._changeAttrOpen(oldVal, newVal);
         }
     }//attributeChangedCallback
 
-    _close (evt) {
```

Comment:
This was renamed to `_onBtnClose` to reduce confusion around intended use (it is a callback to an event listener, not an action method to call at any time). Additionally, the method was moved to the bottom of the file, near the rest of the private functions.